Assets/Scripts/InteractableController.cs

```diff
@@ -40,13 +45,32 @@ void OnTriggerExit2D(Collider2D other)
     // Update is called once per frame
     void Update()
     {
+        //Debug.LogFormat("Text number is {0}", textnumber);
+        textnumber = Random.Range(1, 5);
```

Comment:
This is ok for now, but it can very quickly start to cause performance problems down the road if we don't work to avoid this pattern.

Since the Update() method runs every single frame, we're getting a random number 12 times per frame currently — because this script exists on 12 different game objects. Most computers won't sneeze at that, but imagine if we follow this pattern another few times — it adds up.

We should ideally *only* be getting a new random number when we *know* we want to show text. So we could just move all of this logic into the nested `if` blocks below. So like:

```c#
if (playerIsColliding) {
    if (Input.GetKeyDown(KeyCode.Z) || Input.GetKeyDown("space")) {
        // First, get the random number
        textNumber = Random.Range(1, 5);

        // Next, find the corresponding text value
        if (textNumber == 1) { // We can use the stronger comparison of == here because we know textNumber is an int
            text = text1;
        } else if (textNumber == 2) { // We should use else ifs to avoid checking every if statement ever single time we run down the list
            text = text2;
        } // ... and so on ...

        // Finally, we handle the input
        HandleInput();
    } 
} 
```

In principle, this is fine for now, but one potential headache we could run into down the line is making sure the random number generator's range is equal to the number of text options. So, for example, if some objects had more or less than 4 possible text lines, this would break.

An easy way to fix this would be to create an array of strings, customize that length within the editor, and then we can just get the length of that array and use it as the maximum range value here.

Let's not worry about that for now. But that's something I could help refactor someday!

Reply:
It took a second, but this makes sense to me. I didn't realize it was possible to nest the random roll and text assignment inside the key press before it gets handled. That's so cool! Editing the script to use this method and to use `==` and `elseif`.